core/schema/modulesource.go

```diff
@@ -514,6 +514,106 @@ func (s *moduleSchema) filterUnInstalledDeps(ctx context.Context, bk *buildkit.C
 	return effectiveDependencies, nil
 }
 
+func (s *moduleSchema) applyDepUpdates(ctx context.Context, bk *buildkit.Client, currentDeps []*modules.ModuleConfigDependency, updateList []string, updateAll bool) ([]*modules.ModuleConfigDependency, error) {
+	if len(updateList) == 0 && !updateAll {
+		return currentDeps, nil
+	}
+
+	updatesDepsMap := map[string]parsedRefString{}
+	toBeUpdated := map[string]string{}
+	for _, dep := range updateList {
```

Comment:
Might be worth doing this in an `errgroup`.

Reply:
Thinking more about this. The UX would most likely be user trying to update all the deps (in which case this code is not exercised) or updating one dep at a time. 

so I am thinking if it is worth the complexity to add the `errgroup` here? what do you think?